src/ui/control/navigation_control.js

```diff
@@ -68,7 +68,7 @@ class NavigationControl {
     }
 
     _rotateCompassArrow() {
-        const rotate = `rotate(${this._map.transform.angle * (180 / Math.PI)}deg)`;
+        const rotate = `rotateX(${this._map.transform._pitch * (180 / Math.PI)}deg) rotateZ(${this._map.transform.angle * (180 / Math.PI)}deg)`;
```

Comment:
It would be better to use the `transform#pitch` property instead of the internal variable.

Reply:
Ok, fixed now! 👍